object.go

```diff
@@ -578,77 +613,82 @@ func (obs *obs) Delete(name string) error {
 		return ErrBadObjectMeta
 	}
 
-	// Place a rollup delete marker.
+	// Place a rollup delete marker / publish the meta
 	info.Deleted = true
 	info.Size, info.Chunks, info.Digest = 0, 0, _EMPTY_
-
-	metaSubj := fmt.Sprintf(objMetaPreTmpl, obs.name, sanitizeName(name))
-	mm := NewMsg(metaSubj)
-	mm.Data, err = json.Marshal(info)
-	if err != nil {
+	if err = publishMeta(obs, name, info); err != nil {
 		return err
 	}
-	mm.Header.Set(MsgRollup, MsgRollupSubject)
-	_, err = obs.js.PublishMsg(mm)
+
+	// Purge chunks for the object
+	return purgeChunks(obs, info.NUID)
+}
+
+func doesntExistOrIsDeleted(obs *obs, name string) error {
```

Comment:
Should this be a func on (*obs)?

Reply:
again, lack of go knowledge

Reply:
If you always have a first arg that looks like self, that is a good test.